service/test/cmd/typecheckrpc.go

```diff
@@ -0,0 +1,201 @@
+package main
```

Comment:
Is this checking that the new structs match the old arguments, or that the new RPC methods all contain struct args? Haven't fully studied the code, but could use some documentation anyways. Also, is this intended to be run as part of a CI check to ensure consistent RPC method types (in/out structs)?